src/v2/view-builder/views/phone/EnrollAuthenticatorPhoneView.js

```diff
@@ -1,173 +1,43 @@
-import { _, loc, Model } from 'okta';
-import BaseView from '../../internals/BaseView';
+import { loc } from 'okta';
+import {
+  default as ChallengeAuthenticatorPhoneView,
+  ResendView
+} from './ChallengeAuthenticatorPhoneView';
 import BaseForm from '../../internals/BaseForm';
-import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
-import CountryUtil from '../../../../util/CountryUtil';
-import AuthenticatorEnrollFooter from '../../components/AuthenticatorEnrollFooter';
 
-const Body = BaseForm.extend({
+const EnrollResendView = ResendView.extend(
+  {
+    // To be shown after a timeout
+    className: 'phone-authenticator-enroll--warning hide',
 
-  className: 'phone-authenticator-enroll',
-
-  title () {
-    return loc('oie.phone.enroll.title', 'login');
-  },
-
-  subtitle () {
-    return loc('oie.phone.enroll.subtitle', 'login');
-  },
-
-  render () {
-    BaseForm.prototype.render.apply(this, arguments);
-    const selectedMethod = this.model.get('authenticator.methodType');
-    const phoneField = this.el.querySelector('.phone-authenticator-enroll__phone');
-    const extensionField = this.el.querySelector('.phone-authenticator-enroll__phone-ext');
-
-    if (selectedMethod === 'voice') {
-      if (!phoneField.classList.contains('phone-authenticator-enroll__phone--small')) {
-        phoneField.classList.add('phone-authenticator-enroll__phone--small');
+    handleResendLink () {
+      this.options.appState.trigger('invokeAction', 'currentAuthenticator-resend');
```

Comment:
can create properties in `ResendView` named `resendActionName` hence only need to override this properties.

Reply:
updated